### PR TITLE
Keys generate & derive cleanup, bip44 support for generate

### LINKED ISSUE #600: Refactor key generate API
I believe it would be better to refactor the keys generate service API back to how it was before the key derive functionality: https://github.com/onflow/flow-cli/blob/master/pkg/flowkit/services/keys.go#L56 
We should instead extract this to another service method called `Derive`.

### PR DESCRIPTION
**Pull request by bluesign**
Closes #600

## Description

Will add documentation if this PR looks good. 

For contributor use:

- [ ] Targeted PR against `master` branch
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/flow-cli/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation
- [X] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

**Comment by bluesign** [comment]
Can be reviewed easier when https://github.com/onflow/flow-cli/pull/628 merged. Was not sure how to make the PR tbh.

File: internal/keys/generate.go [hunk 1]
```diff
@@ -57,11 +59,31 @@ func generate(
 		return nil, fmt.Errorf("invalid signature algorithm: %s", generateFlags.KeySigAlgo)
 	}
 
-	privateKey, err := services.Keys.Generate(generateFlags.Seed, "", sigAlgo)
+	//old generation with seed - deprecated
+	if generateFlags.Seed != "" {
```
Comment (sideninja) [inline internal/keys/generate.go:63]: Add a warning to prefer using mnemonic for key generation.

File: pkg/flowkit/keys.go [hunk 2]
```diff
@@ -251,3 +257,80 @@ func (a *HexAccountKey) Validate() error {
...
+}
+
+func (a *Bip44AccountKey) Validate() error {
+
+	if !bip39.IsMnemonicValid(a.mnemonic) {
+		return fmt.Errorf("invalid mnemonic defined for account in flow.json")
+	}
+
+	derivationPath, err := goeth.ParseDerivationPath(a.derivationPath)
+
```
Comment (sideninja) [inline pkg/flowkit/keys.go:309]:
```suggestion
```

File: pkg/flowkit/services/keys.go [hunk 3]
```diff
@@ -52,15 +56,69 @@ func NewKeys(
...
+		if err != nil {
+			return nil, err
+		}
+	}
+	privateKey, err := crypto.DecodePrivateKey(sigAlgo, accountKey.Key)
+	if err != nil {
+		return nil, err
+	}
+	return privateKey, nil
+
```
Comment (sideninja) [inline pkg/flowkit/services/keys.go:105]:
```suggestion
```

File: pkg/flowkit/services/keys.go [hunk 4]
```diff
@@ -52,15 +56,69 @@ func NewKeys(
...
+	if derivationPath == "" {
+		derivationPath = "m/44'/539'/0'/0/0"
+	}
+
+	if !bip39.IsMnemonicValid(mnemonic) {
+		return nil, fmt.Errorf("invalid mnemonic")
+	}
+
+	path, err := goeth.ParseDerivationPath(derivationPath)
+
```
Comment (sideninja) [inline pkg/flowkit/services/keys.go:82]:
```suggestion
```

**Review by sideninja — CHANGES_REQUESTED** [review]
The tests for the new mnemonic keys are needed. Will pass another review after.

**Comment by bluesign** [comment]
@sideninja for the test: expected result key for the path from mnemonic kind of test ?

**Comment by sideninja** [comment]
> @sideninja for the test: expected result key for the path from mnemonic kind of test ?

To test if the derivation works from the provided memonic.

**Comment by bluesign** [comment]
Added 2 tests, one for generic (lilac etc), one for ledger devices.

**Review by sideninja — APPROVED** [review]

**Pull request merged**